Add configuration details from settings.py (#861)
astronomer/astro-sdk: python-sdk/docs/configurations.rst

   [astro_sdk]
   schema = "tmp"

We can modify all databases by one config ``AIRFLOW__ASTRO_SDK__SQL_SCHEMA``. We can also configure the default schema specific to the database type (example: specific to Snowflake, BigQuery, Postgres). If both the default and database-specific schemas are defined, the preference is given to the database-specific value.

.. code:: python

   AIRFLOW__ASTRO_SDK__POSTGRES_DEFAULT_SCHEMA = "postgres_tmp"
   AIRFLOW__ASTRO_SDK__BIGQUERY_DEFAULT_SCHEMA = "bigquery_tmp"
   AIRFLOW__ASTRO_SDK__SNOWFLAKE_DEFAULT_SCHEMA = "snowflake_tmp"
   AIRFLOW__ASTRO_SDK__REDSHIFT_DEFAULT_SCHEMA = "redshift_tmp"

or by updating Airflow's configuration

.. code:: ini

   [astro_sdk]
   postgres_default_schema = "postgres_tmp"
   bigquery_default_schema = "bigquery_tmp"
   snowflake_default_schema = "snowflake_tmp"
   redshift_default_schema = "redshift_tmp"

Configuring the unsafe dataframe storage
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The dataframe are pickled in astro-sdk-python and saved in airflow database. The dataframe is defined by the user and if it is big enough it might potentially break Airflow's metadata. Hence, unsafe dataframe storage should be set to True.
> 

Reviewer: ```suggestion
The dataframes are pickled in astro-sdk-python and saved in the airflow metadata database. Since this dataframe is defined by the user and if it is huge, it might potentially break Airflow's metadata DB by using all the available resources. Hence, unsafe dataframe storage should be set to ``True`` once you are aware of this risk and are OK with it.

```

> The dataframes are pickled in astro-sdk-python and saved in the airflow metadata database. Since this dataframe is defined by the user and if it is huge, it might potentially break Airflow's metadata DB by using all the available resources. Hence, unsafe dataframe storage should be set to ``True`` once you are aware of this risk and are OK with it.

Reviewer: ```suggestion
The dataframes (generated by `dataframe` or `transform` operators) are stored in XCom table using pickling in the Airflow metadata database. Since this dataframe is defined by the user and if it is huge, it might potentially break Airflow's metadata DB by using all the available resources. Hence, unsafe dataframe storage should be set to ``True`` once you are aware of this risk and are OK with it. Alternatively, you could use a Custom XCom backend to store the XCom data
```

.. code:: shell

   AIRFLOW__ASTRO_SDK__DATAFRAME_ALLOW_UNSAFE_STORAGE = True

or by updating Airflow's configuration

.. code:: shell

   [astro_sdk]
   dataframe_allow_unsafe_storage = True

Configuring the storage integration for Snowflake
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
A storage integration is a Snowflake object that stores a generated identity and access management (IAM) entity for your external cloud storage, along with an optional set of allowed or blocked storage locations (Amazon S3, Google Cloud Storage, or Microsoft Azure). Cloud provider administrators in your organization grant permissions on the storage locations to the generated entity. This option allows users to avoid supplying credentials when creating stages or when loading or unloading data.

Read more at: `Snowflake storage integrations <https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html>`_

.. code:: shell

   AIRFLOW__ASTRO_SDK__SNOWFLAKE_STORAGE_INTEGRATION_AMAZON = "aws_integration"
   AIRFLOW__ASTRO_SDK__SNOWFLAKE_STORAGE_INTEGRATION_GOOGLE = "gcp_integration"

or by updating Airflow's configuration

> .. code:: shell

Reviewer: ```suggestion
.. code:: ini
```

   [astro_sdk]
   snowflake_storage_integration_amazon = "aws_integration"
   snowflake_storage_integration_google = "gcp_integration"

Configuring the table autodetect row count
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Following configuration indicates how many file rows should be loaded to infer the table columns types. This defaults to 1000 rows.

> .. code:: shell

Reviewer: ```suggestion
.. code:: ini
```

   AIRFLOW__ASTRO_SDK__LOAD_TABLE_AUTODETECT_ROWS_COUNT = 1000

or by updating Airflow's configuration

> .. code:: shell

Reviewer: ```suggestion
.. code:: ini
```

   [astro_sdk]
   load_table_autodetect_rows_count = 1000

Configuring the RAW SQL maximum response size
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reduce responses sizes returned by aql.run_raw_sql to avoid trashing the Airflow DB if the BaseXCom is used.

.. code:: shell

   AIRFLOW__ASTRO_SDK__RUN_RAW_SQL_RESPONSE_SIZE = 1

or by updating Airflow's configuration

> .. code:: shell

Reviewer: ```suggestion
.. code:: ini
```

   [astro_sdk]
   run_raw_sql_response_size = 1

Configuring the Dataset inlets/outlets
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Astro SDK automatically adds inlets and outlets for all the operators if DATASET is supported (Airflow >=2.4).

> Users can over-ride it on a task level by adding inlets and outlets themselves. However, for users who want to use SDK with Airflow 2.4 and above but do not want to leverage Data-aware scheduling, auto-addition of inlets/outlets would be annoying so this new config setting will provide an escape hatch for such users.

Reviewer 2: ```suggestion
Users can overwrite it on a task level by adding inlets and outlets themselves. However, for users who want to use SDK with Airflow 2.4 and above but do not want to leverage Data-aware scheduling, auto-addition of inlets/outlets would be annoying so this new config setting will provide an escape hatch for such users.
```

> Users can over-ride it on a task level by adding inlets and outlets themselves. However, for users who want to use SDK with Airflow 2.4 and above but do not want to leverage Data-aware scheduling, auto-addition of inlets/outlets would be annoying so this new config setting will provide an escape hatch for such users.

Reviewer: ```suggestion
While users can override it on a task level by adding inlets and outlets, this might be inconvenient for some users who do not want to leverage Data-aware scheduling. Such users can set the following config to `False` to disable auto addition of inlets and outlets
```

.. code:: shell

   AIRFLOW__ASTRO_SDK__AUTO_ADD_INLETS_OUTLETS = True

or by updating Airflow's configuration

> .. code:: shell

Reviewer: ```suggestion
.. code:: ini
```

.. code:: python

>    AIRFLOW__ASTRO_SDK__POSTGRES_DEFAULT_SCHEMA = "postgres_tmp"

Reviewer 3: @sunank200 we should also say we can modify all databases by one config `AIRFLOW__ASTRO_SDK__SQL_SCHEMA` and also the precedence in case `AIRFLOW__ASTRO_SDK__BIGQUERY_DEFAULT_SCHEMA` and `AIRFLOW__ASTRO_SDK__SQL_SCHEMA` is defined.

Author: [Done](https://github.com/astronomer/astro-sdk/pull/861/commits/68f2bf333711b3c05666e56fb61073fe3853cb84)

.. code:: python

   AIRFLOW__ASTRO_SDK__POSTGRES_DEFAULT_SCHEMA = "postgres_tmp"
   AIRFLOW__ASTRO_SDK__BIGQUERY_DEFAULT_SCHEMA = "bigquery_tmp"
   AIRFLOW__ASTRO_SDK__SNOWFLAKE_DEFAULT_SCHEMA = "snowflake_tmp"
   AIRFLOW__ASTRO_SDK__REDSHIFT_DEFAULT_SCHEMA = "redshift_tmp"

or by updating Airflow's configuration

> .. code:: shell

Reviewer: ```suggestion
.. code:: ini
```

   [astro_sdk]
   postgres_default_schema = "postgres_tmp"
   bigquery_default_schema = "bigquery_tmp"
   snowflake_default_schema = "snowflake_tmp"
   redshift_default_schema = "redshift_tmp"

> Configuring the Custom XCom Backends
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> You can also examine Airflow’s configuration:
> 
> .. code:: shell
> 
>    AIRFLOW__CORE__XCOM_BACKEND="tmp"
> 
> or by updating Airflow's configuration
> 
> .. code:: shell
> 
>    [astro_sdk]
>    xcom_backend = "tmp"

Reviewer: This is not SDK specific and done in Airflow -- so we should remove this

```suggestion
```

Configuring the unsafe dataframe storage
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> .. code:: shell

Reviewer: We should add a description on when a user should change this setting/config

Reviewer 3: @sunank200 this setting is more of content from user because the dataframes are pickled and saved in the airflow database, which is a security risk because of pickling and also from a size perspective since we don't control the size of dataframe, if the dataframe is big enough, it can potentially break the Airflow's metadata.

Author: [Done](https://github.com/astronomer/astro-sdk/pull/861/commits/68f2bf333711b3c05666e56fb61073fe3853cb84)

.. code:: shell

   AIRFLOW__ASTRO_SDK__SNOWFLAKE_STORAGE_INTEGRATION_AMAZON = "aws_integration"
   AIRFLOW__ASTRO_SDK__SNOWFLAKE_STORAGE_INTEGRATION_GOOGLE = "gcp_integration"

or by updating Airflow's configuration

.. code:: shell

   [astro_sdk]
   snowflake_storage_integration_amazon = "aws_integration"
   snowflake_storage_integration_google = "gcp_integration"

Configuring the table autodetect row count
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Following configuration indicates how many file rows should be loaded to infer the table columns types. This defaults to 1000 rows.

.. code:: shell

   AIRFLOW__ASTRO_SDK__LOAD_TABLE_AUTODETECT_ROWS_COUNT = 1000

or by updating Airflow's configuration

.. code:: shell

   [astro_sdk]
   load_table_autodetect_rows_count = 1000

Configuring the RAW SQL maximum response size
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reduce responses sizes returned by aql.run_raw_sql to avoid trashing the Airflow DB if the BaseXCom is used.

.. code:: shell

   AIRFLOW__ASTRO_SDK__RUN_RAW_SQL_RESPONSE_SIZE = 1

or by updating Airflow's configuration

.. code:: shell

   [astro_sdk]

>    run_raw_sql_response_size = 1

Reviewer: Can you also add a section about https://github.com/astronomer/astro-sdk/pull/858#issue-1374648365 please

https://github.com/astronomer/astro-sdk/blob/aef1d014de1972c466b80f82422885f04ee5e577/python-sdk/src/astro/settings.py#L40-L43

Author: [Done](https://github.com/astronomer/astro-sdk/pull/861/commits/966322fcaae518de555cfdae4de50df67d3dc924)
